core/api/src/main/java/org/trellisldp/api/BinaryService.java

```diff
@@ -34,9 +36,10 @@
      *
      * @param metadata the binary metadata
      * @param stream the content
+     * @param headers HTTP request headers
```

Comment:
I would simply call this "hints". I can _easily_ imagine writing some Java code against the core and using this param for some totally non-HTTP metadata.